### PR TITLE
Forming commits url if its git repo for Jenkins commits

### PR DESCRIPTION
**Pull request by kasubram**

### CODE REVIEW

File: src/Agent.Worker/Release/Artifacts/JenkinsArtifact.cs [hunk 1]
```diff
@@ -271,7 +279,10 @@ private Change ConvertCommitToChange(IExecutionContext context, JToken token)
             context.Output(StringUtil.Format("Getting changeSet associated with build {0} ", jobId));
             string commitsUrl = StringUtil.Format("{0}/job/{1}/{2}/api/json?tree=number,result,changeSet[items[commitId,date,msg,author[fullName]]]", artifactDetails.Url, artifactDetails.JobName, jobId);
             var commitsResult = await DownloadCommitsJsonContent(context, commitsUrl, artifactDetails, "$.changeSet.items[*]");
-            return commitsResult.Select(x => ConvertCommitToChange(context, x));
+
+            string rootUrl;
+            bool gitRepo = isGitRepo(context, artifactDetails, jobId, out rootUrl);
```
Comment (GitHubSriramB) [inline src/Agent.Worker/Release/Artifacts/JenkinsArtifact.cs]: bool isGitRepo?

File: src/Agent.Worker/Release/Artifacts/JenkinsArtifact.cs [hunk 2]
```diff
@@ -293,26 +304,86 @@ private Change ConvertCommitToChange(IExecutionContext context, JToken token)
...
+            if (!string.IsNullOrEmpty(result))
+            {
+                executionContext.Debug($"result will be filtered with {jsonPath}");
+                return ParseToken(result, jsonPath);
             }
 
             return new List<JToken>();
         }
 
+        private bool isGitRepo(IExecutionContext executionContext, JenkinsArtifactDetails artifactDetails, int jobId, out string rootUrl)
```
Comment (GitHubSriramB) [inline src/Agent.Worker/Release/Artifacts/JenkinsArtifact.cs]: IsGitRepo?

File: src/Agent.Worker/Release/Artifacts/JenkinsArtifact.cs [hunk 3]
```diff
@@ -293,26 +304,86 @@ private Change ConvertCommitToChange(IExecutionContext context, JToken token)
...
+                    executionContext.Debug($"Parsed repo result {repoKind}");
+
+                    if (!string.IsNullOrEmpty(repoKind) && repoKind.Equals(GitRepoName, StringComparison.OrdinalIgnoreCase))
+                    {
+                        executionContext.Debug("Its a git repo, checking if it has root url");
+                        var rootUrlResult = ParseToken(repoResult.ToString(), "$.actions[?(@.remoteUrls)]");
+                        if (rootUrlResult != null && rootUrlResult.Any())
+                        {
+                            var resultDictionary = JsonConvert.DeserializeObject<Dictionary<string, JToken>>(rootUrlResult.First().ToString());
+                            if (resultDictionary.ContainsKey(RemoteUrlsKey) && resultDictionary[RemoteUrlsKey].Any())
```
Comment (GitHubSriramB) [inline src/Agent.Worker/Release/Artifacts/JenkinsArtifact.cs]: is this check documented as part of Jenkins?

File: src/Test/L0/Worker/Release/JenkinsArtifactL0.cs [hunk 4]
```diff
@@ -80,7 +80,7 @@ public sealed class JenkinsArtifactL0
 
                 var artifact = new JenkinsArtifact();
                 artifact.Initialize(tc);
-                string expectedUrl = $"{details.Url}/job/{details.JobName}/{details.EndCommitArtifactVersion}";
+                string expectedUrl = $"{details.Url}/job/{details.JobName}/{details.EndCommitArtifactVersion}/api/json?tree=number,result,changeSet[items[commitId,date,msg,author[fullName]]]";
```
Comment (GitHubSriramB) [inline src/Test/L0/Worker/Release/JenkinsArtifactL0.cs]: how is this test impacted by this change?

File: src/Agent.Worker/Release/Artifacts/JenkinsArtifact.cs (hunk 1 above)
Comment (kasubram) [inline src/Agent.Worker/Release/Artifacts/JenkinsArtifact.cs]: Done

File: src/Agent.Worker/Release/Artifacts/JenkinsArtifact.cs (hunk 2 above)
Comment (kasubram) [inline src/Agent.Worker/Release/Artifacts/JenkinsArtifact.cs]: Done

File: src/Test/L0/Worker/Release/JenkinsArtifactL0.cs (hunk 4 above)
Comment (kasubram) [inline src/Test/L0/Worker/Release/JenkinsArtifactL0.cs]: there are two request made, one to check if its git repo another to download the commits url, the earlier url was very generic.. it has just jobname and jobId in the url. So moq showed it ran twice.. we should more specifically check, if the commits download url has been called.

File: src/Agent.Worker/Release/Artifacts/JenkinsArtifact.cs (hunk 3 above)
Comment (kasubram) [inline src/Agent.Worker/Release/Artifacts/JenkinsArtifact.cs]: Jenkins documentation does not have much explanation on each of the attribute on its REST api results. I could see this repo kind property coming up in the REST api result for all the repo type. Hence used it.

**Pull request merged**